### PR TITLE
[Content update] src/development/tools/devtools/memory.md

### PR DESCRIPTION
**Pull request by AlexV525**
Improve the document by updating semantics line breaks and spacings, correcting typos, etc.

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/development/tools/devtools/memory.md [hunk 1]
```diff
@@ -17,41 +17,48 @@ memory at a given moment.
...
 ## Allocations and tracking
 
-Monitoring all allocations involes you directly interacting with DevTools
+Monitoring all allocations involves you directly interacting with DevTools
 and your application to isolate a short period of time that you are
 interested in knowing how many objects were allocated, how many bytes
 were allocated, or tracking all the places in your code where a particular
 class is allocated. This information is available under the "Allocations"
-tab of Memory profiler and his a fairly fast with less overhead than using
+tab of Memory profiler, and he's a fairly fast with less overhead than using
```
Comment (domesticmouse) [inline src/development/tools/devtools/memory.md:61]: @sfshaza2 i suspect your wordsmithing is required here

**Review by domesticmouse — COMMENTED** [review]
Ptal @sfshaza2

File: src/development/tools/devtools/memory.md (hunk 1 above)
Comment (sfshaza2) [reply src/development/tools/devtools/memory.md:61]: Oh dear. I just scanned the whole page and, yes, this is in serious need of some wordsmithing. I plan to do that in another PR. Thanks, @AlexV525, for bringing this to my attention!!!

Comment (AlexV525) [reply src/development/tools/devtools/memory.md:61]: This article exists for a long time, while I doing the improvement, I found many emotional sentences that can be improved too.

**Comment by sfshaza2** [comment]
I'm creating a massive PR of this file. Please don't merge this PR. I'm leaving it open as a reminder.

**Comment by sfshaza2** [comment]
Now that I've pushed the new PR, I'm closing this one.

**Pull request closed without merging**